Mathlib/Data/List/Basic.lean

```diff
@@ -2559,6 +2567,75 @@ theorem foldlRecOn_nil {C : β → Sort*} (op : β → α → β) (b) (hb : C b)
   rfl
 #align list.foldl_rec_on_nil List.foldlRecOn_nil
 
+private lemma middle_xYz_left {x₁ x₂ z₁ z₂ : List α} {Y₁ Y₂ : α}
+    (together : x₁ ++ [Y₁] ++ z₁ = x₂ ++ [Y₂] ++ z₂) (longer : x₂.length < x₁.length) :
+    Y₂ ∈ x₁ := by
+  have middle := congr_fun (congr_arg get? together) x₂.length
+  rw [
+    append_assoc x₂,
+    get?_append_right x₂.length.le_refl,
+    Nat.sub_self,
+    singleton_append,
+    get?_cons_zero,
+    append_assoc x₁,
+    get?_append longer,
+  ] at middle
+  exact get?_mem middle
+
+private lemma todo_refactor {a b c d : ℕ}
+    (total : a + b = c + d) (a_lt_c : a < c) :
+    d < b := by
+  by_contra contr
+  rw [not_lt] at contr
+  have impossi : a + b < c + d
+  · exact add_lt_add_of_lt_of_le a_lt_c contr
+  have impossineq : a + b ≠ c + d
+  · exact Nat.ne_of_lt impossi
+  exact impossineq total
+
+lemma match_xYz {x₁ x₂ z₁ z₂ : List α} {Y₁ Y₂ : α} (notin_x : Y₂ ∉ x₁) (notin_z : Y₂ ∉ z₁) :
+    x₁ ++ [Y₁] ++ z₁ = x₂ ++ [Y₂] ++ z₂ ↔ x₁ = x₂ ∧ Y₁ = Y₂ ∧ z₁ = z₂ := by
```

Comment:
Lemma names should not reference the names of their variables. This should be named something like `append_singleton_append_inj_of_nmem`

Reply:
Thank you!